### DISCUSSION TITLE
Does Yellow work with php 7.0/7.1?

### DISCUSSION THREAD

**Discussion by gdippe** [datenstrom/community#264] [Ask a question] [answered]
I have set up a new test server with Apache 2.4 and php 7.0 (initially 7.1).
Now this message appear: 
_Datenstrom Yellow requires PHP mbstring extension!_ but it is loaded as far as I can tell.
I compared php.ini for php 5.6 and 7.0 and there are quite a few differences when it comes to mbstring. But I'm stuck and haven't been able to find a solution.
Any help would be much appreciated.

php -m
[PHP Modules]
Core
ctype
date
dom
filter
hash
iconv
json
libxml
mbstring
mysqli
mysqlnd
pcre
PDO
pdo_sqlite
Phar
posix
Reflection
session
SimpleXML
SPL
sqlite3
standard
tokenizer
xml
xmlreader
xmlwriter
Zend OPcache

[Zend Modules]
Zend OPcache

and partly

php -i 
mbstring

Multibyte Support => enabled
Multibyte string engine => libmbfl
HTTP input encoding translation => disabled
libmbfl version => 1.3.2
oniguruma version => 5.9.6

mbstring extension makes use of "streamable kanji code filter and converter", which is distributed under the GNU Lesser General Public License version 2.1.

Multibyte (japanese) regex support => enabled
Multibyte regex (oniguruma) version => 5.9.6

Directive => Local Value => Master Value
mbstring.detect_order => no value => no value
mbstring.encoding_translation => Off => Off
mbstring.func_overload => 0 => 0
mbstring.http_input => no value => no value
mbstring.http_output => no value => no value
mbstring.http_output_conv_mimetypes => ^(text/|application/xhtml\+xml) => ^(text/|application/xhtml\+xml)
mbstring.internal_encoding => no value => no value
mbstring.language => neutral => neutral
mbstring.strict_detection => Off => Off
mbstring.substitute_character => no value => no value

**Comment by BsNoSi** [comment, marked as answer]
I use several instances of Yellow *always* using PHP 7.x, this works well. I have to admit, that PHP 7.x. was a major reason for switching to Yellow.

I know that this does not solve your issue, but you can delete php 7. x from your list of possible reasons.

**Comment by markseuffert** [comment]
Yes, PHP 7 is supported. My guess is that the Apache's PHP configuration is the problem.

Which PHP version do you use exactly?
Which operating system do you use?

**Comment by hysonlee** [comment]
I'm sure Yellow works fine with PHP 7+Apache, also the latest Yellow release. you can check the message header of my demo site: https://yellow.hyson.tk

**Comment by gdippe** [comment]
I use FreeBSD 11, PHP version 7.0.20.

**Comment by markseuffert** [comment]
Check whether your `mbstring` extension is available in the Apache web server.

I am afraid to say that I am not familiar with FreeBSD, it was years ago when I used it.

On your average Unix system (Debian/Ubuntu) you can enable `mbstring` like this:

```
sudo apt-get install php-mbstring php7.0-mbstring
sudo phpenmod mbstring 
sudo service apache2 restart   
```

Hope this helps. 😄

**Comment by gdippe** [comment]
Thanks for quick answers/suggestions. On FreeBSD it's `sudo pkg install apache24` So it is quite similar to different Linuxes.

**Comment by markseuffert** [comment]
Contact me, if you want me to have a closer look.

**Comment by gdippe** [comment]
Thank you! I'll have vacation starting tomorrow and continue when I'm back.

**Comment by markseuffert** [comment]
Do you need help or can we close the issue?

**Comment by gdippe** [comment]
You can close this issue.

Thank you!

Günther

> On 20 Sep 2017, at 14:58, Mark Seuffert <notifications@github.com> wrote:
> 
> Do you need help or can we close the issue?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/datenstrom/yellow/issues/264#issuecomment-330843186>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AaFC5VvavU4IWzFubf5MXdn0kvYMNPNOks5skQvmgaJpZM4OOvu8>.
> 

Günther Dippe
Blåsutgatan 1
414 56 Göteborg
Sweden

+46 (0)709 781 935
skype: gdippe
web: gdfoto.se
